### PR TITLE
Fixes for tests in service recording

### PR DESCRIPTION
**Pull request by MichaelOrlov**
- Relates https://github.com/ros2/rosbag2/pull/1480

List of fixes:
- Address flakiness in newly added rosbag2_transport tests
- Use wait_for_srvice_to_be_ready() instead of the check_service_ready()
- Add rosbag2_test_common::wait_until_condition(..)
- Rewrite get_service_info tests to be more deterministic
- Renames in the get_service_info tests
- Use `std::filesystem` instead of `rcpputils::fs` functions
  - Rationale: Deprecation of the rcpputils::fs in future
See https://github.com/ros2/rcpputils/issues/164 for details
- Avoid extra metadata readout in `ros2 bag info` verb with `--verbose`
- Add info_end_to_end test with `--verbose` parameter

### CODE REVIEW

**Comment by MichaelOrlov** [comment]
@Barry-Xu-2018 Please review and merge to your branch if you have no concerns or requests for changes.

File: ros2bag/ros2bag/verb/info.py [hunk 1]
```diff
@@ -47,14 +47,14 @@ def _is_service_event_topic(self, topic_name, topic_type) -> bool:
         return True
 
     def main(self, *, args):  # noqa: D102
-        m = Info().read_metadata(args.bag_path, args.storage)
-        if args.topic_name:
-            for topic_info in m.topics_with_message_count:
-                if not self._is_service_event_topic(topic_info.topic_metadata.name,
-                                                    topic_info.topic_metadata.type):
-                    print(topic_info.topic_metadata.name)
+        if args.verbose:
```
Comment (Barry-Xu-2018) [inline ros2bag/ros2bag/verb/info.py:50]: How about to output a warning while users set '-t' and '-v' at the same time ?
 
 ```python
 if args.topic_name and args.verbose :
     print("Warning！You have set both the '-t' and '-v' parameters. The '-t' parameter will be ignored.")
```

**Comment by Barry-Xu-2018** [comment]
@MichaelOrlov 

Thank you for your assistance.   
I checked each one, and they look good to me.    
About info.py, how about to output a warning while users pass '-t' and '-v' at the same time.

**Comment by Barry-Xu-2018** [comment]
I merged them. 

> About info.py, how about to output a warning while users pass '-t' and '-v' at the same time.

If you think it is okay, I will add it in ros2/rosbag2#1480.

File: ros2bag/ros2bag/verb/info.py (hunk 1 above)
Comment (MichaelOrlov) [reply ros2bag/ros2bag/verb/info.py:50]: @Barry-Xu-2018 Sorry for my late response. Yeah, with warning will be even better. :+1:

**Comment by MichaelOrlov** [comment]
Sorry for my late response. Yeah, with warning will be even better. 👍

File: ros2bag/ros2bag/verb/info.py (hunk 1 above)
Comment (Barry-Xu-2018) [reply ros2bag/ros2bag/verb/info.py:50]: Okay. I added a new commit https://github.com/ros2/rosbag2/pull/1480/commits/ab62249b08e2ba497b1928eaacad8876437c8610.

**Pull request merged**